Block/Adminhtml/System/Config/Field/Installment.php

```diff
@@ -11,48 +11,55 @@
 
 namespace Adyen\Payment\Block\Adminhtml\System\Config\Field;
 
-class Installment extends \Magento\Framework\View\Element\Html\Select
+use Magento\Framework\View\Element\Html\Select;
+
+class Installment extends Select
 {
-    /**
-     * Options
-     *
-     * @var array
-     */
-    protected $options = [
-        '1' => '1x',
-        '2' => '2x',
-        '3' => '3x',
-        '4' => '4x',
-        '5' => '5x',
-        '6' => '6x',
-        '7' => '7x',
-        '8' => '8x',
-        '9' => '9x',
-        '10' => '10x',
-        '11' => '11x',
-        '12' => '12x',
-        '13' => '13x',
-        '14' => '14x',
-        '15' => '15x',
-        '16' => '16x',
-        '17' => '17x',
-        '18' => '18x',
-        '19' => '19x',
-        '20' => '20x',
-        '21' => '21x',
-        '22' => '22x',
-        '23' => '23x',
-        '24' => '24x'
-    ];
+    private function getSourceOptions(): array
+    {
+        return [
+            ['label' => '1x', 'value' => '1'],
+            ['label' => '2x', 'value' => '2'],
+            ['label' => '3x', 'value' => '3'],
+            ['label' => '4x', 'value' => '4'],
+            ['label' => '5x', 'value' => '5'],
+            ['label' => '6x', 'value' => '6'],
+            ['label' => '7x', 'value' => '7'],
+            ['label' => '8x', 'value' => '8'],
+            ['label' => '9x', 'value' => '9'],
+            ['label' => '10x', 'value' => '10'],
+            ['label' => '11x', 'value' => '11'],
+            ['label' => '12x', 'value' => '12'],
+            ['label' => '13x', 'value' => '13'],
+            ['label' => '14x', 'value' => '14'],
+            ['label' => '15x', 'value' => '15'],
+            ['label' => '16x', 'value' => '16'],
+            ['label' => '17x', 'value' => '17'],
+            ['label' => '18x', 'value' => '18'],
+            ['label' => '19x', 'value' => '19'],
+            ['label' => '20x', 'value' => '20'],
+            ['label' => '21x', 'value' => '21'],
+            ['label' => '22x', 'value' => '22'],
+            ['label' => '23x', 'value' => '23'],
+            ['label' => '24x', 'value' => '24']
+        ];
+    }
 
-    /**
-     * Sets name for input element
-     *
-     * @param string $value
-     * @return $this
-     */
-    public function setInputName($value)
+    public function _toHtml(): string
+    {
+        if (!$this->getOptions()) {
+            $this->setOptions($this->getSourceOptions());
+        }
```

Comment:
Can we move that to the class constructor?

Reply:
I would rather keep it here because that class is an extension of `Magento\Framework\View\Element\Html` and this method will be called specifically. What do you think?

Reply:
Magento instantiates this block internally, and this function is called at the point of rendering the template so having the options set here give us a bit more control over how and when this is run